src/mysql/index.ts

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env node
+
+import { Server } from "@modelcontextprotocol/sdk/server/index.js";
+import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js";
+import {
+  CallToolRequest,
+  CallToolRequestSchema,
+  ListResourcesRequest,
+  ListResourcesRequestSchema,
+  ListToolsRequest,
+  ListToolsRequestSchema,
+  ReadResourceRequest,
+  ReadResourceRequestSchema,
+} from "@modelcontextprotocol/sdk/types.js";
+import mysql from "mysql2/promise";
+
+const server = new Server(
+  {
+    name: "example-servers/mysql",
+    version: "0.1.0",
+  },
+  {
+    capabilities: {
+      resources: {},
+      tools: {},
+    },
+  },
+);
+
+const args = process.argv.slice(2);
+if (args.length === 0) {
+  console.error("Please provide a database URL as a command-line argument");
+  process.exit(1);
+}
+
+const databaseUrl = args[0];
+
+const resourceBaseUrl = new URL(databaseUrl);
+resourceBaseUrl.protocol = "mysql:";
+resourceBaseUrl.password = "";
+
+const pool = mysql.createPool(databaseUrl);
+
+const SCHEMA_PATH = "schema";
+
+server.setRequestHandler(ListResourcesRequestSchema, async (_request: ListResourcesRequest) => {
+  const [rows] = await pool.query(
+    "SELECT table_name FROM information_schema.tables WHERE table_schema = DATABASE()",
+  );
+  
+  return {
+    resources: (rows as any[]).map((row) => ({
+      uri: new URL(`${row.TABLE_NAME}/${SCHEMA_PATH}`, resourceBaseUrl).href,
+      mimeType: "application/json",
+      name: `"${row.TABLE_NAME}" database schema`,
+    })),
+  };
+});
+
+server.setRequestHandler(ReadResourceRequestSchema, async (request: ReadResourceRequest) => {
+  const resourceUrl = new URL(request.params.uri);
+
+  const pathComponents = resourceUrl.pathname.split("/");
+  const schema = pathComponents.pop();
+  const tableName = pathComponents.pop();
```

Comment:
We should probably be a bit more strict about the exact format here, e.g. test if there are more than two path components